Source/WebCore/rendering/RenderBlockFlow.cpp

```diff
@@ -861,6 +861,14 @@ void RenderBlockFlow::layoutBlockChildren(bool relayoutChildren, LayoutUnit& max
 
 void RenderBlockFlow::trimBlockEndChildrenMargins()
 {
+    auto trimSelfCollapsingChildDescendants = [&](RenderBox& child) {
+        ASSERT(child.isSelfCollapsingBlock());
+        for (auto itr = RenderIterator<RenderBox>(&child, child.firstChildBox()); itr; itr = itr.traverseNext()) {
```

Comment:
isn't it just `for (auto& childBox : childrenOfType<RenderBox>(*this))`?

Reply:
Yes, it does appear so. Changed